app/controllers/api/v1/books_controller.rb

```diff
@@ -0,0 +1,27 @@
+module Api
+  module V1
+    class BooksController < ApiController
+      include Wor::Paginate
+      include Brita
+
+      filter_on :author, type: :string
+      filter_on :genre, type: :string
+      filter_on :title, type: :string
+      filter_on :description, type: :string
+
+      before_action :authenticate_user!
+
+      # Summary: List books with filters ang paginated
+      def index
+        books = filtrate(Book.all)
+        render_paginated(books, each_serializer: BookSerializer)
+      end
+
+      # Summary: Show a single book json by id
+      def show
+        return render(plain: 'Not Found', status: :not_found) unless Book.exists?(params['id'])
+        render(json: Book.find(params['id']), serializer: ShowBookSerializer)
```

Comment:
This is cool, but there is another way to do it that keeps it cleaner.
`.find` raises a `ActiveRecord::RecordNotFound` exception if there is no record with that id, so you could catch that with a `rescue_from ActiveRecord::RecordNotFound, with: :record_not_found` (in the ApiController, so you could reuse it)
Finally, you have to define the record_not_found function, that could be:
def record_not_found
render json: { error: 'Record not found' }, status: :not_found
end

All of this leaves you to just use the Entity.find directly in every controller, without having to worry if it exists or not, because it's been already handled

Reply:
Oks, It's good because handle the specific error but I have the habit of validate over handler.